editor/animation_track_editor_plugins.cpp

```diff
@@ -101,47 +101,67 @@ void AnimationTrackEditColor::draw_key_link(int p_index, float p_pixels_sec, int
 	int font_size = get_theme_font_size("font_size", "Label");
 	int fh = (font->get_height(font_size) * 0.8);
 
+	fh /= 3;
+
 	int x_from = p_x + fh / 2 - 1;
 	int x_to = p_next_x - fh / 2 + 1;
-	fh /= 3;
+	x_from = MAX(x_from, p_clip_left);
+	x_to = MIN(x_to, p_clip_right);
+
+	int y_from = (get_size().height - fh) / 2;
 
 	if (x_from > p_clip_right || x_to < p_clip_left) {
 		return;
 	}
 
-	Color color = get_animation()->track_get_key_value(get_track(), p_index);
-	Color color_next = get_animation()->track_get_key_value(get_track(), p_index + 1);
+	Vector<Color> color_samples;
+	color_samples.append(get_animation()->track_get_key_value(get_track(), p_index));
 
-	if (x_from < p_clip_left) {
-		float c = float(p_clip_left - x_from) / (x_to - x_from);
-		color = color.lerp(color_next, c);
-		x_from = p_clip_left;
-	}
+	if (get_animation()->track_get_type(get_track()) == Animation::TYPE_VALUE) {
```

Comment:
Is this condition necessary? Can color track be a different track than TYPE_VALUE?

Reply:
The condition is necessary because there are `value` specific calls  (e.g. `value_track_interpolate`).
Ideally, it should work for every animation type where interpolation at specific time makes sense (currently, only the `value` and `bezier` type). But since Animation doesn't expose a generic equivalent of  `value_track_interpolate` and colors in `bezier` are split in their component, I decided to support only `value`.

Reply:
I mean, it should be impossible to create a color track of other type, so this condition would never evaluate to false anyways. But maybe safety check just in case is ok 🤔

Reply:
Honestly, I'd keep it because it has a good fallback. But I agree that it's probably useless currently.